### PR TITLE
[rllib] fix incorrect critic loss in TD3

### LINKED ISSUE #10653: [rllib] DDPG TWIN_Q LOSS ERROR
The TD3 loss according to OpenAI Spinning up is:
L1

L2

They implement this as you would expect:
````
# MSE loss against Bellman backup
loss_q1 = ((q1 - backup)**2).mean()
loss_q2 = ((q2 - backup)**2).mean()
loss_q = loss_q1 + loss_q2
````

The rllib implementation is off and I ran into training issues when using it.

Tensorflow:
https://github.com/ray-project/ray/blob/39c598bab0929f007a49daa7262a7c9ee6492221/rllib/agents/ddpg/ddpg_tf_policy.py#L174-L184

Pytorch:
https://github.com/ray-project/ray/blob/5851e893eee491e569fe1421314172bd5b37d583/rllib/agents/ddpg/ddpg_torch_policy.py#L124-L134

The error is on line 177 and 127 respectively where the td_error and twin_td_error are added together and then squared a couple lines later. Removing that line would bring the implementation in line with the spinningup reference implementation. This worked for me to improve training on my problem. They also do not multiply by the constant (0.5) but I would not expect that to make much of a difference either way.

### PR DESCRIPTION
**Pull request by mvindiola1**
## Why are these changes needed?

DDPG critic loss function incorrect when using TD3 options.

## Related issue number

Closes #10653

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by ericl** [comment]
@michaelzhiluo can you check the math here?

**Comment by michaelzhiluo** [comment]
Based on the paper, this is the correct loss function change ^^. You should not add both td_error and twin_td_error since it already adds both later in the code.

**Review by michaelzhiluo — APPROVED** [review]

**Comment by ericl** [comment]
@mvindiola1 looks like you also need to fix the DDPG unit test: https://travis-ci.com/github/ray-project/ray/jobs/384619773#L1725

**Comment by mvindiola1** [comment]
Sorry about that. I had checked ray/rllib/agents/ddpg/tests/test_td3.py and did not see anything relevant in there but I did not check test_ddpg.py. Anyway, I made the required change there and the tests passed locally on my machine.

**Review by ericl — APPROVED** [review]

**Comment by ericl** [comment]
Merged, thanks!
